api/src/main/java/org/openmrs/module/mksreports/dataset/evaluator/ObsSummaryRowDataSetDefinitionEvaluator.java

```diff
@@ -58,8 +58,8 @@ public DataSet evaluate(DataSetDefinition dataSetDefinition, EvaluationContext c
 	 */
 	protected List<Integer> getPatientsWithObs(EvaluationContext context) {
 		SqlQueryBuilder queryBuilder = new SqlQueryBuilder();
-		queryBuilder.append(" select o.person_id from obs o  inner join patient p on o.person_id = p.patient_id  "
-		        + "where o.voided = false and p.voided = false and concept_id = :questionConceptId "
+		queryBuilder.append("select o.person_id from obs o  inner join patient p on o.person_id = p.patient_id  "
+		        + "where o.voided = false and p.voided = false and concept_id IN (:questionConceptIds) "
```

Comment:
@ivange94 both `diagnosisList` and `questionConceptIds` are indeed parameters of the dataset definition (see [here](https://github.com/ivange94/openmrs-module-mksreports/blob/6920a7ad8f25419575e53af0cc709d4006085584/api/src/main/java/org/openmrs/module/mksreports/reports/OutpatientConsultationReportManager.java#L143) for instance). That's good, but where is the reporting framework directed to understand what the _value_ of this parameter is?

This parameter is not mapped straight through from a parent definition, and is not set explicitly anywhere. Am I missing something?

Reply:
The parameters are set at runtime in the test. See https://github.com/ivange94/openmrs-module-mksreports/blob/TLC-122/api/src/test/java/org/openmrs/module/mksreports/reports/OutpatientConsultationReportManagerTest.java#L98-L104. As for the parameter mapping I'm not really sure how they work as my code seems to have been working without them.